scripts/source-maintenance.sh

```diff
@@ -40,6 +40,26 @@ fi
 COPYRIGHT_YEARS=`date +"1996-%Y"`
 echo "s/1996-2[0-9]+ The Squid Software Foundation and contributors/${COPYRIGHT_YEARS} The Squid Software Foundation and contributors/g" >>boilerplate_fix.sed
 
+updateifchanged()
+{
+original="$1"
+updated="$2"
+message="$3"
+
+if test -e "${original}" -a -e "${updated}"; then
+	md51=`cat "${original}" | tr -d "\n \t\r" | ${MD5}`;
+	md52=`cat "${updated}" | tr -d "\n \t\r" | ${MD5}`;
+	if test "${md51}" != "${md52}" ; then
+		echo "NOTICE: File ${original} changed: ${message}"
+		mv "${updated}" "${original}"
+	else
+		rm -f "${updated}"
+	fi
+else
+	echo "BUG: invalid filename '$1' or '$2'. Skipped update for ${message}."
```

Comment:
done

Reply:
As to your question. Two reasons;

* The MD5 tools produce output regardless of whether a file exists. So we get a false-positive detection of change if the .new does not exist.

* There is no point doing all the work of sub-shell creation and MD5 calculation if either file is non-existent. This being the lowest level of post-processing it adds up to a significant portion of the script running time doing useless work.

Reply:
I consider this change request addressed, thank you. IIRC, there is another (more general) change request about quitting on errors where that more general discussion can take place.

As for those "two reasons", I have doubts about them -- we would not get to updateIfChanged at all if the source file did not exist (unstaged changes are prohibited and the plugin would have failed) or if the output file was not created (leading to the short-circuited `plugin && updateIfChanged` command failure) -- but perhaps it is not important to reach an agreement on this interpretation. I am mentioning these doubts here primarily because they are a good illustration why quitting on the first error is a good default for most scripts -- it eliminates or reduces these and many similar concerns/uncertainties/risks. But that discussion is for a different change request...